crates/rolldown_loader_utils/src/binary_to_esm.rs

```diff
@@ -1,20 +1,26 @@
 use rolldown_common::Platform;
 
-pub fn binary_to_esm(base64: &str, platform: Platform, runtime_resource_id: &str) -> String {
+pub fn binary_to_esm(
+  base64: &str,
+  platform: Platform,
+  runtime_resource_id: &str,
+) -> anyhow::Result<String> {
   let to_binary = match platform {
     Platform::Node => "__toBinaryNode",
     _ => "__toBinary",
   };
-  [
-    "import {",
-    to_binary,
-    "} from '",
-    runtime_resource_id,
-    "'; export default ",
-    to_binary,
-    "('",
-    base64,
-    "')",
-  ]
-  .concat()
+  Ok(
```

Comment:
You don't need to add Result here.

Reply:
Ok. It may be my obsession on uniforming :-), but due to the performance, and understandability, etc. I will delete it. Thanks for figuring it out!

Reply:
But it seems that the `Result` for the `base64` function is necessary, otherwise may lead to `clippy::match_same_arms`.
Both `DataUrl` and `Base64` requires the base64 because I treat them same, while the `DataUrl` has the "preprocess" step reading the source.